### ISSUE TITLE
HomeAssistant?

### ISSUE THREAD

**Issue by anandlar** [sittzon/shinemonitor_api#3]
Have you integrated shinemonitor to Home assistant?

**Comment by sittzon** [comment]
> Have you integrated shinemonitor to Home assistant?

Haven't had time to do that, but it would be great to have!

**Comment by anandlar** [comment]
I would be nice. Sadly I'm not a programmer. I'm trying to understand what to do, but fail bigtime :-D

**Comment by me-2k** [comment]
@anandlar did you get the swagger REST-API up and running?
If so, you should add something like this to your home assistant configuration.yaml file.
Replace 192.168.101.3:8001 with your swagger IP.

`
rest:
  - authentication: basic
    method: GET
    scan_interval: 60
    verify_ssl: false
    resource: https://192.168.101.3:8001/EnergySummary
    sensor:
      - name: "Solar Production Today"
        device_class: energy
        state_class: total_increasing
        unit_of_measurement: "kWh"
        value_template: "{{ float(value_json.today) }}"
      - name: "Solar Production Month"
        device_class: energy
        state_class: total_increasing
        unit_of_measurement: "kWh"
        value_template: "{{ float(value_json.month) }}"
      - name: "Solar Production Year"
        device_class: energy
        state_class: total_increasing
        unit_of_measurement: "kWh"
        value_template: "{{ float(value_json.year) }}"
      - name: "Solar Production Total"
        device_class: energy
        state_class: total_increasing
        unit_of_measurement: "kWh"
        value_template: "{{ float(value_json.total) }}"
  - authentication: basic
    method: GET
    scan_interval: 60
    verify_ssl: false
    resource: https://192.168.101.3:8001/EnergyNow
    sensor:
      - name: "Solar Power Now"
        device_class: power
        state_class: measurement
        unit_of_measurement: "W"
        value_template: "{{ float(value_json.energy) }}"
`

**Comment by anandlar** [comment]
No, I have not got it working, but it would be nice. Sadly it is a little bit over my head :-(

**Comment by MysterX83** [comment]
Hi i've create a mqtt producer based on this code:
https://github.com/MysterX83/shinemonitor_mqtt

please feel free to copy the code and use it for use.

**Comment by anandlar** [comment]
No, I haven't done that yet. Quite unsure how to do it (I'm not a
developer). Thank you for your suggestion below :-)
I know how to modify the YAML file, so that will not be a problem .

Anders

Den sön 12 juni 2022 08:46me-2k ***@***.***> skrev:
